internal/server/iigointernal/legislature.go

```diff
@@ -109,35 +111,39 @@ func (l *legislature) RunVote(ruleID string, clientIDs []shared.ClientID) voting
 	ruleVote.SetVotingIslands(clientIDs)
 
 	ruleVote.GatherBallots(iigoClients)
-	//TODO: log of vote occurring with ruleID, clientIDs
-	//TODO: log of clientIDs vs islandsAllowedToVote
-	//TODO: log of ruleID vs s.RuleToVote
+
+	variablesToCache := []rules.VariableFieldName{rules.IslandsAllowedToVote}
+	valuesToCache := [][]float64{{float64(len(clientIDs))}}
+	l.monitoring.addToCache(l.SpeakerID, variablesToCache, valuesToCache)
+
+	variablesToCache = []rules.VariableFieldName{rules.SpeakerRuleProposal, rules.PresidentRuleProposal}
+	valuesToCache = [][]float64{{float64(len(ruleID))}, {float64(len(l.ruleToVote))}}
```

Comment:
I'm so confused here. You are passing the length of the string ruleID and l.ruleToVote. What happens if there are two different strings of ruleID that have the same length value?

Reply:
Is it possible to create some tests so I can see this in action?

Reply:
oh shit, yes this is a mistake. ruleid and l.ruletovote are of type string so i think this is a combination of my ide suggesting a change and me blindly following it. I will redo this rule. Doesn't make a lot of sense, sorry!

Reply:
Ah okay. No probs.

Reply:
Basically i just need to store something so that later the judge can check if they were equal so i've changed it to store a bool now. Should be a touch more logical now!